src/fsharp/service/service.fs

```diff
@@ -687,75 +709,90 @@ type BackgroundCompiler(legacyReferenceResolver, projectCacheSize, keepAssemblyC
     /// Fetch the check information from the background compiler (which checks w.r.t. the FileSystem API)
     member __.GetBackgroundCheckResultsForFileInProject(filename, options, userOpName) =
         reactor.EnqueueAndAwaitOpAsync(userOpName, "GetBackgroundCheckResultsForFileInProject", filename, fun ctok -> 
-          cancellable {
+            cancellable {
             let! builderOpt, creationErrors = getOrCreateBuilder (ctok, options, userOpName)
             match builderOpt with
-            | None -> 
+            | None ->
                 let parseResults = FSharpParseFileResults(creationErrors, None, true, [| |])
                 let typedResults = FSharpCheckFileResults.MakeEmpty(filename, creationErrors, reactorOps, keepAssemblyContents)
                 return (parseResults, typedResults)
             | Some builder -> 
                 let! (parseTreeOpt, _, _, untypedErrors) = builder.GetParseResultsForFile (ctok, filename)
                 let! tcProj = builder.GetCheckResultsAfterFileInProject (ctok, filename)
+
+                let tcInfo, tcInfoOptional = tcProj.TcInfoFull ctok
```

Comment:
How do you determine whether to space separate functions or have them right next to each other?  I struggle with this too. I wondered why line 742, 733 and 735 are empty lines.

Reply:
In this case, the call `TcInfoFull` is really important to see clearly as it can do a lot of computation depending on if `enablePartialTypeChecking` in `FSharpChecker` is set to true.

Reply:
Lines 733/735 - the spaces could be removed.